opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -1405,4 +1405,26 @@ int BlackoilWellModelGeneric::numLocalNonshutWells() const
     return well_container_generic_.size();
 }
 
+
+void BlackoilWellModelGeneric::initInjMult() {
+    for (auto& well : this->well_container_generic_) {
+        if (well->isInjector() && well->wellEcl().getInjMultMode() != Well::InjMultMode::NONE) {
+            const auto& ws = this->wellState().well(well->indexOfWell());
+            const auto& perf_data = ws.perf_data;
+            if ((this->prev_inj_multipliers_.count(well->name())) == 0 ) {
+                this->prev_inj_multipliers_[well->name()] = std::vector<double>(perf_data.size(), 1.0);
```

Comment:
This is fractionally better written as
```C++
this->prev_inj_multipliers_[well->name()].assign(perf_data.size(), 1.0)
```
You can even go further and write all of this, including the check for existence, as
```C++
this->prev_inj_multipliers_
    .emplace(std::piecewise_construct,
             std::forward_as_tuple(well->name()),
             std::forward_as_tuple(perf_data.size(), 1.0));
```
This statements depends on the fact that `unordered_map<>::emplace()` does not insert into the map if the key already exists.  That way you don't have to look up `well->name()` twice.  Function [`forward_as_tuple()`](https://en.cppreference.com/w/cpp/utility/tuple/forward_as_tuple) is declared in the `<tuple>` header.

Reply:
At the end, I will use the following code. It is more readable to me and it only searches once. 

```c++
auto &values = this->prev_inj_multipliers_[well->name()];
if (values.empty()) {
    values.assign(perf_data.size(), 1.0);
}
```